### PR TITLE
Share Kit 4.0.0

### PR DESCRIPTION
**Pull request by JaceHensley**
## Ultimate Problem
We need to render a QR code on desktop and a button on mobile so BloomID users can tap to verify on their mobile device and have it open the bloom mobile app.

## Solution
- Create `renderRequestElement` which will render either a QR code or `<a />` depending on the client

## Media

verify with bloom

## 4.0.0

With 4.0.0 share-kit is now licensed under Apache-2.0

**BREAKING**

- All deprecated functions are removed
- React is no longer a dependency
- Migration
  - `generateRequestQRCOde` => `renderRequestElement`
  - `manageRequestQRCode` => `renderRequestElement`
  - `RequestQRCode` => `RequestElement` (from `share-kit-react` coming soon!!)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,23 +17,22 @@
     "dist/**"
   ],
   "scripts": {
-    "build": "yarn clean && ./bin/build",
+    "build": "npm run clean && ./bin/build",
     "clean": "rimraf dist",
-    "prepack": "yarn build",
-    "prepare": "yarn build",
+    "prepack": "npm run build",
```
Comment (edhedges) [inline package.json]: Do we need both `prepare` and `prepack` to do the same thing? Will this end up building the package twice upon publishing?

File: src/elements/renderRequestButton.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  anchor.target = '_blank'
+  anchor.rel = 'norefferer noopener'
+
+  anchor.appendChild(renderStyle(id))
+  anchor.appendChild(renderLockIcon(id))
+  anchor.appendChild(renderTextAndLogo(id))
+
+  config.container.appendChild(anchor)
+
+  return {
```
Comment (edhedges) [inline src/elements/renderRequestButton.ts]: So, by generating an id and applying it to the `update` and `remove` returned to the client, we facilitate the markup management for the client? I like it! 💯

File: src/elements/renderRequestQRCode.ts [hunk 3]
```diff
@@ -192,8 +192,8 @@ const makeEyeBit = (ctx: CanvasRenderingContext2D, info: CellInfo, connectionTyp
   }
 }
 
-const generateRequestQRCode = (canvas: HTMLCanvasElement, data: RequestData, options: Partial<Options>) => {
-  const defaultedOptions = {...defaultOptions, ...options}
+const drawCanvas = (canvas: HTMLCanvasElement, data: RequestData, qrOptions?: Partial<QROptions>) => {
+  const defaultedOptions = {...defaultOptions, ...qrOptions}
```
[the remark this answers is not in the document]
Comment (JaceHensley) [reply src/elements/renderRequestQRCode.ts]: Correct, it defaults any value not specified in `qrOptions`. Open to different names

File: src/elements/renderRequestButton.ts (hunk 2 above)
Comment (JaceHensley) [reply src/elements/renderRequestButton.ts]: Yup! consumer doesn't need to keep track of the id to update/remove the element. Just have to hold on to the return value

File: src/renderRequestElement.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+const renderRequestElement = (config: {
+  container: HTMLElement
+  requestData: RequestData
+  qrOptions?: Partial<QROptions>
+  shouldRenderButton?: ShouldRenderButton
+}): RequestElementResult => {
+  if (config.shouldRenderButton === undefined) {
+    config.shouldRenderButton = parsedResult => {
+      const isMobile = parsedResult.platform.type === 'mobile' || parsedResult.platform.type === 'tablet'
+      const isIOS = () => parsedResult.os.name === 'iOS'
```
Comment (edhedges) [inline src/renderRequestElement.ts]: Is there a reason why this isn't a boolean? I'd pick either boolean or function to be consistent if not.

File: package.json (hunk 1 above)
Comment (JaceHensley) [reply package.json]: I removed the `prepack` script

**Pull request merged**